### PR TITLE
Introduce UndefinedBehaviorSanitizer (UBSan)

### PR DESCRIPTION
**Pull request by glevkovich**
* **Problem Overview**  
* UBSan is a fast undefined behavior detector.
UBSan modifies the program at compile-time to catch various kinds of
undefined behavior during program execution

See https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html.

To enable UBSan 
1) In dockerize mode: make build CONCORD_BFT_CMAKE_UBSAN=TRUE
2) Inside container, or native mode via CMake, set UNDEFINED_BEHAVIOR_CHECK as TRUE.

Currently, no Gihub workflow is defined, and no actions are done to make it part of any CI run.

* **Testing Done**  

  NA

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -80,9 +81,19 @@ string(APPEND CMAKE_CXX_FLAGS " -Wbuiltin-macro-redefined")
...
+# At most, one sanitizer should be defined from below ALL_SANITIZER_OPTS
+set(COUNTER 0)
+set(ALL_SANITIZER_OPTS LEAKCHECK;THREADCHECK;UNDEFINED_BEHAVIOR_CHECK)
+foreach(option IN LISTS ALL_SANITIZER_OPTS)
+  if(${option})
+    MATH(EXPR COUNTER "${COUNTER}+1")
+  endif()
+endforeach()
+if(${COUNTER} GREATER 1)
+  message(FATAL_ERROR "More than one sanitizer was chosen!")
```
Comment (WildFireFlum) [inline CMakeLists.txt:94]: Please give an error message which shows the chosen options in addition to their amount.

**Review by WildFireFlum — APPROVED** [review]
Thank you for adding this.
I had one comment regarding the error message you display.

**Comment by eyalrund** [comment]
Did it find anything for you?

**Comment by glevkovich** [comment]
`Did it find anything for you?`
@eyalrund , I havn't checked the whole codebase, it will take ages.
Ran only one apollo test suite, and yes - it did find some issues.

**Review by eyalrund — APPROVED** [review]

**Pull request merged**